ipn/ipnlocal/local.go

```diff
@@ -4624,6 +4632,13 @@ func (b *LocalBackend) applyPrefsToHostinfoLocked(hi *tailcfg.Hostinfo, prefs ip
 	hi.ShieldsUp = prefs.ShieldsUp()
 	hi.AllowsUpdate = envknob.AllowsRemoteUpdate() || prefs.AutoUpdate().Apply.EqualBool(true)
 
+	// count routes without exit node routes
+	for _, route := range hi.RoutableIPs {
+		if route.Bits() != 0 {
+			metricAdvertisedRoutes.Add(1)
```

Comment:
I think you'd want to increment a local variable, and then set a gauge at the end of the loop.

Perhaps there is an existing test for subnet routers that we can use to confirm that the metric exposes correct value?